### PR TITLE
render readme html scrapped via the server

### PR DESCRIPTION
**Pull request by getneil**
DEMO: https://www.loom.com/share/71fab2e24ffe442da15102d62d76867c

- we probably need to make the backend ai generation part more reliable, some packages still dont have an html readme because they dont have a public git repo
- only works for those packages with generated readme
- if readme.html failed from s3, gui manually fetches data from github or gitlab
- show images for relative path src

### CODE REVIEW

**Comment by getneil** [comment]
how do we handle if readme has been updated by the package maintainer? atm, readme will only rebuild on release of new build.

**Comment by getneil** [comment]
added issues:
- https://github.com/teaxyz/gui/issues/782
- https://github.com/teaxyz/gui/issues/783

or just do all correctly remotely which i prefer
https://github.com/teaxyz/api/issues/81

**Review by ABevier — APPROVED** [review]
🚢

File: svelte/src/components/markdown/markdown.svelte [hunk 1]
```diff
@@ -24,7 +23,15 @@
     link: Link
   };
 
-  $: html = source.type === "rst" ? rst2html(source.data) : "";
+  $: html = source.type === "rst" ? rst2html(source.data) : source.data;
+
+  // TODO: support gitlab too
+  const changeSrcIfNeeded = (element: HTMLElement | Element) => {
+    const src = element.getAttribute("src");
+    if (src?.startsWith("/")) {
```
Comment (ryanml) [inline svelte/src/components/markdown/markdown.svelte:31]: I'm not sure this single condition will cover all the bases for relative URLs. I had found most examples of a relative path in the Openssl, lefthook, and xcodegen libraries. Among those three you get srcs like:

```
assets/imgs/foo.png
/assets/imgs/foo.png
./assets/imgs/foo.png
```

I had begun working on a remedy to this issue yesterday before the s3 solution was mentioned/implemented, and a regular expression covered the bases for relative paths we'd run in to in READMEs. Something like:
`const isRelativePath = src?.match(/^[^\/]+\/[^\/].*$|^\/[^\/].*$/i)`

Comment (ryanml) [reply svelte/src/components/markdown/markdown.svelte:31]: Note: This regex could probably be simplified, as it covers conditions we won't likely run in to as well

Comment (getneil) [reply svelte/src/components/markdown/markdown.svelte:31]: @ryanml ill add that thank you, i missed out on that hehehe

**Pull request merged**